ubuntudesign/git-mirror-service
# Don't require secret file

Make sure that it is possible to use the server unsecured, so if you haven't created
`.secret` then you don't need the `?secret=` parameter.
## QA

``` bash
echo "something" > .secret  # Create a secret
./wsgi.py                   # Run the server
```

Visit http://127.0.0.1:9052/?origin=something&destination=something-else. You should see:

```
Missing one of `secret`, `origin` or `destination`
```

Now try without the secret:

``` bash
rm -rf .secret  # Make sure there's no secret file
./wsgi.py       # Run the server
```

Visit http://127.0.0.1:9052/?origin=something&destination=something-else. You should now see in the browser:

```
A server error occurred.  Please contact the administrator.
```

And in your shell:

```
CalledProcessError: Command '['/home/robin/projects/git-mirror-service/copy-repository.py', 'something', 'something-else']' returned non-zero exit status 1
```

This is expected, as `something` and `something-else` aren't real repositories. We're not testing that bit. But it shows that the server gets as far as trying to copy the repository.
